distributed/tests/test_worker_client.py

```diff
@@ -315,3 +315,16 @@ def test_submit_different_names(s, a, b):
         assert fut > 0
     finally:
         yield c.close()
+
+
+@gen_cluster(client=True)
+async def test_task_unique_groups_scatter(c, s, a, b):
+    """ This test ensure that tasks are correctly deleted when using scatter/submit
+    """
+    a = await c.scatter([0, 1], hash=True)
+    x = await c.submit(sum, a)
+    del a
+    del x
+    b = await c.scatter([1, 2], hash=True)
+    y = await c.submit(sum, b)
+    assert s.task_prefixes["sum"].states["memory"] == 1
```

Comment:
You could also try `y.key in b.data and s.tasks[y.key].status == "memory"`

There is a slight chance that gc doesn't trigger for a while, so this prefix number could be 2

Reply:
In general, one learns to stop assuming that gc will behave sensibly :)

Reply:
Thanks